Adding Microsoft //Build 2019 scenario (#95)
dotnet/spark: scenarios/csharp/GitHubLovesSparkDotNet/README.md

In the **Github 💖 .NET for Apache Spark** app, we dive into the world of
analyzing GitHub meta-data using .NET for Apache Spark. While the analysis
is simple, it conveys a key point of how easy it is to get started with
analyzing large datasets using .NET.

# Table of Contents

- [Introduction](#introduction)
- [Technology Stacks](#technology-stacks)
- [Pre-requisites](#pre-requisites)
- [Running the App](#running-the-app)
- [Discussion](#discussion)

> - [Insights](#insights)

Reviewer: Remove ?  I don't see an insights section below.

Author: I'll add this now. Thank you!

- [Venues](#venues)
- [Acknowledgements](#acknowledgements)

# Introduction

[GHTorrent](http://ghtorrent.org/) monitors the [Github public event time line](https://api.github.com/events). 
For each event, it retrieves its contents and their dependencies, exhaustively. It 
then stores the data in a MySQL database. 

> **Disclaimer**: This app is purely for demonstration purposes. And more 
importantly, this was not written by a seasoned Data Scientist so please 
avoid deriving *any* conclusions from the results :) Of course, please feel 
free to open PRs to improve and make this app more interesting.

# Technology Stacks

This app utilizes and consequently familiarizes you with the following technology:

  - [Azure Databricks](https://azure.microsoft.com/en-us/services/databricks/)
  - [Azure Data Lake Storage Gen2](https://azure.microsoft.com/en-us/services/storage/data-lake-storage/)
  - [Databricks CLI](https://docs.databricks.com/user-guide/dev-tools/databricks-cli.html)
  - [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/)
  - [Visual Studio Community Edition](https://visualstudio.microsoft.com/vs/community/)

> 

Reviewer 2: Consider adding a link to getting started tutorial so that user knows they have the option to run locally.
```suggestion
[Running Spark Apps Locally](https://dotnet.microsoft.com/learn/data/spark-tutorial/intro)

```

# Pre-requisites

## Accounts & Cluster Creation

> Make sure you have [created an Azure Databricks cluster and attached an Azure Data Lake Gen2 filesystem](https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-use-databricks-spark) before you proceed.

Reviewer 2: Is there a reason the user needs to use Azure Databricks specifically? If so, I'd mention why. If not, I'd make this step broader. 
```suggestion
Make sure you can either [run .NET for Apache Spark apps locally](https://dotnet.microsoft.com/learn/data/spark-tutorial/intro) or in the cloud. For help with cloud deployment, you can try [creating an Azure Databricks cluster and attaching an Azure Data Lake Gen2 filesystem](https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-use-databricks-spark).
```

Author: I give this demo using Azure Databricks in my talks. Users are free to use any cloud provider. I see having another scenario that shows HDI specific things. For now, I'm marking this resolved but please feel free to reopen if you disagree.

## Dataset Preparation

  1. Download the latest MySQL CSV dataset from [GHTorrent Downloads page](http://ghtorrent.org/downloads.html)
  2. On your machine, untar (use [7-zip for Windows](https://www.7-zip.org/)) the dataset

>      ```

Reviewer: ```suggestion
     ```shell
```

     cd <path-to-mysql-csv-dump>
     tar xvzf mysql-2019-05-01.tar.gz
     ```
  3. Use [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/) to upload the dataset into your storage account

# Running the App

>   1. Change the constants in `Program.cs` to reflect your account details

Reviewer 2: Give the user an example of what their filled-in account details should look like.

  2. Follow the [publish & deploy instructions for Azure Databricks](https://github.com/dotnet/spark/tree/master/deployment#databricks)

> 

Reviewer 2: Like I had mentioned in feedback on the general README, I'd break this down a bit more to explain what's going on (why we create a Spark Session for each app, reading into a dataframe, using functional programming/Spark SQL/UDFs to analyze data).

# Discussion

Work-in-progress

# Venues

> This app was presented at Microsoft //Build 2019:

Reviewer: ```suggestion
This app was presented at [Microsoft //Build 2019](https://www.microsoft.com/en-us/build):
```
